### PR TITLE
xq: embed build metadata

### PR DESCRIPTION
**Pull request by sibprogrammer**
- [*] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [*] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [*] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [*] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [*] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [*] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/xq.rb [hunk 1]
```diff
@@ -18,7 +18,16 @@ class Xq < Formula
...
   def install
-    system "go", "build", *std_go_args(ldflags: "-s -w -X main.version=#{version}")
+    ENV["CGO_ENABLED"] = "0"
+    build_date = Time.now.strftime("%FT%TZ")
+    ldflags = %W[
+      -s -w
+      -X main.commit=#{Utils.git_head}
+      -X main.version=#{version}
+      -X main.date=#{build_date}
+    ]
```
Comment (bayandin) [inline Formula/xq.rb:28]:
```suggestion
    ldflags = %W[
      -s -w
      -X main.commit=#{Utils.git_head}
      -X main.version=#{version}
      -X main.date=#{time.rfc3339}
    ]
```

Comment (sibprogrammer) [reply Formula/xq.rb]: time.rfc3339 looks like a reference to a commit date, not a build date. But let it be :)

Comment (bayandin) [reply Formula/xq.rb]: Yep, we prefer this for [build reproducibility](https://reproducible-builds.org/)

File: Formula/xq.rb [hunk 2]
```diff
@@ -1,10 +1,10 @@
 class Xq < Formula
   desc "Command-line XML beautifier and content extractor"
   homepage "https://github.com/sibprogrammer/xq"
-  url "https://github.com/sibprogrammer/xq/archive/refs/tags/v0.0.8.tar.gz"
-  sha256 "9ab6d0fd2ef3e1ca6c74fec8a466c20d65fd5330eaf75d16a22bf878c5500493"
+  url "https://github.com/sibprogrammer/xq.git",
+    tag:      "v0.0.8",
+    revision: "b249d1a8470c10d950e404f926794bd5ba7f7c3a"
```
Comment (carlocab) [inline Formula/xq.rb:6]:
```suggestion
      tag:      "v0.0.8",
      revision: "b249d1a8470c10d950e404f926794bd5ba7f7c3a"
```

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**